FXyz-Importers/src/main/java/org/fxyz3d/importers/obj/MtlReader.java

```diff
@@ -54,87 +53,143 @@
     private String baseUrl;
 
     public MtlReader(String filename, String parentUrl) {
-        baseUrl = parentUrl.substring(0,parentUrl.lastIndexOf('/')+1);
+        baseUrl = parentUrl.substring(0, parentUrl.lastIndexOf('/') + 1);
         String fileUrl = baseUrl + filename;
-        try {
-            URL mtlUrl = new URL(fileUrl);
-            ObjImporter.log("Reading material from filename = " + mtlUrl);
-            read(mtlUrl.openStream());
-        } catch (FileNotFoundException ex) {
-            System.err.println("No material file found for obj. ["+fileUrl+"]");
-        } catch (IOException ex) {
-            ex.printStackTrace();
+        try (Stream<String> line = Files.lines(Paths.get(new URI(fileUrl)))) {
+            line.map(String::trim).filter(l -> !l.isEmpty() && !l.startsWith("#")).forEach(this::parse);
```

Comment:
One line is nice, of course, but I like each operation of the stream in one line